src/cpp/client/channel_cc.cc

```diff
@@ -116,10 +117,11 @@ internal::Call Channel::CreateCall(const internal::RpcMethod& method,
     } else if (!host_.empty()) {
       host_str = host_.c_str();
     }
-    grpc_slice method_slice = SliceFromCopiedString(method.name());
+    grpc_slice method_slice =
+        SliceFromArray(method.name(), strlen(method.name()));
     grpc_slice host_slice;
     if (host_str != nullptr) {
-      host_slice = SliceFromCopiedString(host_str);
+      host_slice = SliceFromArray(host_str, strlen(host_str));
```

Comment:
note `host_str` is defined locally. So it either comes from `context->authority()` which returns a `string` or from `host_` which is a `const string`. I guess using `SliceFromCopiedString` makes sense? With changes around line 115 of course.

If it helps, note we are a friend of `context` and we can access `context->authority_` directly.

Reply:
Ah yes, great point. I changed this part accordingly.